ducktape/tests/runner.py

```diff
@@ -203,6 +203,12 @@ def teardown_single_test(self, teardown_services=True):
             exceptions.append(e)
             self.log(logging.WARN, "Error freeing nodes: %s" % e.message + "\n" + traceback.format_exc(limit=16))
 
+        try:
+            self.current_test_context.close_logger()
+        except BaseException as e:
```

Comment:
I was being cautious. Cursory internet investigation does not show any errors coming from closing a file though.